iothub/device/src/Transport/Amqp/AmqpUnit.cs

```diff
@@ -1013,19 +1013,22 @@ private async Task OpenTwinSenderLinkAsync(AmqpIotSession amqpIotSession, string
             }
         }
 
-        private void OnDesiredPropertyReceived(AmqpMessage responseFromService, string correlationId, IotHubClientException ex = default)
+        private async Task OnDesiredPropertyReceivedAsync(AmqpMessage responseFromService, string correlationId, IotHubClientException ex = default)
         {
             if (Logging.IsEnabled)
-                Logging.Enter(this, responseFromService, nameof(OnDesiredPropertyReceived));
+                Logging.Enter(this, responseFromService, nameof(OnDesiredPropertyReceivedAsync));
 
             try
             {
-                _twinMessageListener?.Invoke(responseFromService, correlationId, ex);
+                if (_twinMessageListener != null)
```

Comment:
I mean, we didn't choose to before. You make a good case for it, but I'd probably go with a Debug.Assert instead so we only find it during testing.

Not really related to this PR, though.

I had to change this because you can't use the null conditional operator on an async call.